src/lib/environment/mod.rs

```diff
@@ -534,6 +534,10 @@ pub(crate) fn setup_cwd(cwd: Option<&str>) {
                 "CARGO_MAKE_CARGO_HOME",
                 &home::cargo_home_with_cwd(directory_path).ok(),
             );
+            envmnt::set_or_remove(
+                "CARGO_MAKE_CRATE_TARGET_TRIPLE",
+                &crateinfo::crate_target_triple(None).or_else(|| rust_info::get().target_triple),
```

Comment:
rust_info::get() is short but its an expensive call.
i wonder if we can in the mod call get rust info first and pass it to get crate info to enable sharing of data and prevent that expensive call. wdyt?

Reply:
Sounds perfectly doable, will look into it in a bit.

Reply:
I solved it with just getting `CARGO_MAKE_RUST_TARGET_TRIPLE`, instead of polluting all these functions with more arguments I thought.

Reply:
but that means we are counting on it that it is invoked first. if its not a param, there is no 'contract' to enforce it. it may fail later on quietly if i changed the order for some dumb reason.

Reply:
Hm ... 🤔
Should I move it away from `setup_cwd` then? I mean is there a more fitting place for it?

Reply:
I did a little workaround, but it works now.